src/main/services/csfloat/csfloatSocket.ts

```diff
@@ -0,0 +1,566 @@
+import { JsonTradeoffer } from "./../../models/types";
+import { EventEmitter } from "node:events";
+import CSFloatClient from "./csfloatClient";
+import {
+  ICSFloatSocketEvents,
+  ITradeFloat,
+  IUpdateErrors,
+  TradeOffersAPIOffer,
+} from "./interfaces/csfloat.interface";
+import {
+  EStatusTradeCSFLOAT,
+  ETradeOfferStateCSFloat,
+} from "./enums/cs-float.enum";
+import { sleepAsync } from "@doctormckay/stdlib/promises";
+import { minutesToMS } from "../../../shared/helpers";
+import {
+  IGetTradeOffersResponse,
+  IHistoryPingBody,
+} from "./interfaces/fetch.interface";
+import { AppId } from "./enums/steam.enum";
+import CEconItem from "steamcommunity/classes/CEconItem.js";
+
+export declare interface CSFloatSocket {
+  emit<U extends keyof ICSFloatSocketEvents>(
+    event: U,
+    ...args: Parameters<ICSFloatSocketEvents[U]>
+  ): boolean;
+
+  on<U extends keyof ICSFloatSocketEvents>(
+    event: U,
+    listener: ICSFloatSocketEvents[U]
+  ): this;
+
+  once<U extends keyof ICSFloatSocketEvents>(
+    event: U,
+    listener: ICSFloatSocketEvents[U]
+  ): this;
+}
+
+export class CSFloatSocket extends EventEmitter {
+  private connected: boolean;
+  private _csFloatClient: CSFloatClient;
+  private steamIDBase64: string;
+  private statusExtension = false;
+
+  constructor(_CSFloatClient: CSFloatClient, steamIDBase64: string) {
+    super();
+    this._csFloatClient = _CSFloatClient;
+    this.steamIDBase64 = steamIDBase64;
+  }
+
+  public disconnect() {
+    this.connected = false;
+    this.emit("stateChange", false);
+    return;
+  }
+
+  public async connect(): Promise<void> {
+    this.connected = true;
+    await this.registerLoops();
+  }
+
+  private async registerLoops(): Promise<void> {
+    while (this.connected) {
+      try {
+        this.emit("stateChange", true);
+
+        const tradesOffers = await this.getTradesOffers();
+        const tradesInQueue = await this._csFloatClient.getTrades(
+          EStatusTradeCSFLOAT.QUEUED
+        );
+
+        await this.autoAcceptTrades(tradesInQueue, tradesOffers);
+
+
+        const tradesInPending = await this._csFloatClient.getTrades(
+          EStatusTradeCSFLOAT.PENDING
+        );
+
+        await this.verificationsLoop(tradesInPending);
+        await this.sendOffer(tradesInPending);
+        await this.extensionLoop(
+          tradesInPending,
+          tradesOffers
+        );
+      } catch (err) {
+        this.emit("error", err);
+      }
+      await sleepAsync(minutesToMS(3));
+    }
+
+    this.disconnect();
+  }
+
+  private async extensionLoop(
+    tradesInPending: ITradeFloat[],
+    tradeOffers: IGetTradeOffersResponse
+  ): Promise<void> {
+    let errors: IUpdateErrors;
```

Comment:
Start the variable with a empty obj, like the original extension code.
```suggestion
    let errors: IUpdateErrors = {};
```